### PR TITLE
Attempt to fix minimist CVE

### PR DESCRIPTION
**Pull request by vim-usds**
# Purpose 

TL;DR - A vulnerability is blocking FE builds. This vulnerability requires us to update some libraries. This is a major change to the FE. Please see the UI changes below

### Longer explanation

<details>

A recent FE build continues to [fail](https://github.com/usds/justice40-tool/runs/5653482972?check_suite_focus=true). The reason was the [minimist package was polluting the prototype in v1.2.5](https://github.com/advisories/GHSA-xvch-5gv4-984h).

Even though this PR seems to build (minimist patched to v1.2.6), some say that it seems to still [exist](https://snyk.io/test/npm/minimist/1.2.6). It's possible that the low severity isn't getting blocked by GItHub yet and will possibly become a high severity one day. 🤷 .

We are using older versions of core libraries used on the FE. When these core libraries jump _major_ versions, the previous major versions tend to get less support. We are now on older _major_ versions of

- react
- gatsby
- maplibre
- react-map-gl

These are the core libraries of the CEJST FE.

Given that this minimist dependency needed to be bumped, it might be a good opportunity to update some of our core libraries as well. This PR attempts to do just that. Updating to the latest and greatest is usually not recommended. Instead a step-by-step approach could be better:

1. Update to the latest _minor_ version
2. Update to the oldest _major_ version
3. Update to the latest _major_ version

</details>

### UI Changes due to updating our core libraries
- the zoom/territory shortcuts have changed
- the search icon now uses the default search
- the color of the launch icon in survey button

### Code base changes
<details>

- udpate npm to 8.5.5
- update node to 14.17.5
- update all minor versions to latest
- minimist to 1.2.6 (could still be CVE)
- set docker frontend to use cdn by default
- update all dependencies
- update survey button launch svg color
- update to zoom / territory buttons
- search icon changed
- update Feature type from maplibre-gl
- use default search icon

</details>

## Testing
I am requesting that the entire team take part in this as its a significant change to the system:

### QA tester - Vim

Tests
- [x] Do all pages load?
- [x] Do all pages load on mobile?
- [x] Do all the links work?
- [x] Does the map load?
- [x] Does the map zoom in/ zoom out?
- [x] Does the territory shortcuts work?
- [x] Does search work?
- [x] Does selecting and deselecting a tract work?
- [x] Does the methodology page links work?
- [ ] Does favicon change on prod?

### QA tester - Kameron

Tests
- [x] Do all pages load?
- [ ] Do all pages load on mobile?
- [x] Do all the links work?
- [x] Does the map load?
- [ ] Does the map zoom in/ zoom out?
- [ ] Does the territory shortcuts work?
- [ ] Does search work?
- [ ] Does selecting and deselecting a tract work?
- [ ] Does the methodology page links work?

### QA tester - Jorge

Tests
- [x] Do all pages load?
- [x] Do all pages load on mobile?
- [x] Do all the links work?
- [x] Does the map load?
- [x] Does the map zoom in/ zoom out?
- [x] Does the territory shortcuts work?
- [x] Does search work?
- [x] Does selecting and deselecting a tract work?
- [x] Does the methodology page links work?

### QA tester - Katherine

Tests
- [ ] Do all pages load?
- [ ] Do all pages load on mobile?
- [ ] Do all the links work?
- [ ] Does the map load?
- [ ] Does the map zoom in/ zoom out?
- [ ] Does the territory shortcuts work?
- [ ] Does search work?
- [ ] Does selecting and deselecting a tract work?
- [ ] Does the methodology page links work?

### QA tester - Beth

Tests
- [x] Do all pages load?
- [x] Do all pages load on mobile?
- [x] Do all the links work?
- [x] Does the map load?
- [x] Does the map zoom in/ zoom out?
- [x] Does the territory shortcuts work?
- [x] Does search work?
- [x] Does selecting and deselecting a tract work?
- [x] Does the methodology page links work?

### QA tester - Emma

Tests
- [x] Do all pages load?
- [x] Do all pages load on mobile?
- [x] Do all the links work?
- [x] Does the map load?
- [x] Does the map zoom in/ zoom out?
- [x] Does the territory shortcuts work?
- [x] Does search work?
- [x] Does selecting and deselecting a tract work?
- [x] Does the methodology page links work?

### QA tester - Lucas

Tests
- [x] Do all pages load?
- [x] Do all pages load on mobile?
- [x] Do all the links work?
- [x] Does the map load?
- [x] Does the map zoom in/ zoom out?
- [x] Does the territory shortcuts work?
- [x] Does search work?
- [x] Does selecting and deselecting a tract work?
- [x] Does the methodology page links work?

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ services:
     environment:
     # See the client readme for more info on environment variables: 
     # https://github.com/usds/justice40-tool/blob/main/client/README.md
-      DATA_SOURCE: local
+      DATA_SOURCE: cdn
```
Comment (esfoobar-usds) [inline docker-compose.yml:36]: Why is `cdn` set as the data source here?

**Review by esfoobar-usds — CHANGES_REQUESTED** [review]
Just wanna make sure about `docker-compose`

File: docker-compose.yml (hunk 1 above)
Comment (esfoobar-usds) [reply docker-compose.yml:36]: We want whoever checks out the codebase to point to the local tiles, which is now generating more quickly.

Comment (vim-usds) [reply docker-compose.yml:36]: The frontend Dockerfile was designed to be able to have the tiles supplied either locally (by running a full etl run) or by getting it from the CDN. This allows the developer to set this.

https://github.com/usds/justice40-tool/blob/main/client/README.md#environment-variables

Has this changed since we determined this requirement? (seems like ages ago we were talking about this when we were first trying to get docker going)

Comment (vim-usds) [reply docker-compose.yml:36]: Before we launched I was providing support to some of the folks from EPIC. They requested that it be set to the CDN to allow for the fastest way to get the app running. They believed that most people wont need the pipeline running locally. Maybe things have changed? We can discuss this more during discussion hour tmrw.

What are the advantages of running locally that could overcome the cost a longer build time?

**Comment by katherinedm-usds** [comment]
image

**Comment by katherinedm-usds** [comment]
Image from iOS
I noticed this on mobile but no blockers

File: docker-compose.yml (hunk 1 above)
Comment (esfoobar-usds) [reply docker-compose.yml:36]: I agree there's value of a faster startup time with the CDN option. However there are two issues with this approach:

1. It doesn't make sense then that the Data-Pipeline docker instance is even built
2. For data folks, they might want to see how score is locally built, where the tiles are stored and modify the algorithm.

I don't think we should change the whole docker flow on this PR, let's create a new ticket where we enable the ability for a one line `docker run usds/justice40-map` and add this to the README, where users that just want to see the map without building the score, can do so.

Comment (vim-usds) [reply docker-compose.yml]: @jorge - agreed and thanks for explaining!
